iFixit/react-commerce
# Playwright Migration: Migrate first batch of Cypress tests to Playwright

## Description

We will slowly port over the `Cypress` tests to `Playwright`. We will start with the `product` tests so we have some tests to work with while addressing #1114.

## CR Notes:

The Playwright tests are pretty much the same as the Cypress tests.  Of course, there were some necessary changes like using Playwright syntax and how we select elements on the page. 

## QA Notes:

- [ ] Confirm running the Playwright tests will pass locally
	-  ⚠️ You might need to run `npx playwright install` to install the browsers
	- Run `(cd frontend && pnpm playwright:run)` 
- [ ] Stress test the Playwright tests by using the following command
	- `for i in {1..5}; do (cd frontend && pnpm playwright:run --trace on --retries 2); done` 
- [ ] Confirm the Playwright tests execute the same tasks as the Cypress tests
	-  You can add the following diff and then the following command to record videos of Playwright's interactions and then compare that to Cypress's interactions
		```diff
		diff --git a/frontend/playwright.config.ts b/frontend/playwright.config.ts
		index a87beadc..0d92ca1b 100644
		--- a/frontend/playwright.config.ts
		+++ b/frontend/playwright.config.ts
		@@ -48,6 +48,7 @@ const config: PlaywrightTestConfig = {
		 
		       /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
		       trace: 'on-first-retry',
		+      video: 'on',
		 
		       viewport: { width: 1400, height: 1000 },
		```
		- `(cd frontend && pnpm playwright:run --project=chromium)` 

Connects: #1113
